R/README.Rmd

```diff
@@ -0,0 +1,132 @@
+```{r test 2, echo=TRUE}
+execute(h, 'x:1+1') #assign x hopefully to 2
+execute(h, 'x') # get back the value
+```
+
+# Checks
+## Atoms from kdb to R
+
+As per [Q for mortals](http://code.kx.com/q4m3/2_Basic_Data_Types_Atoms/) kdb uses the following basic types:
+
+```{r atoms 0, echo=FALSE, warning=FALSE, message=FALSE}
+# install data.table
+if(! 'data.table' %in% rownames(installed.packages())) install.packages('data.table')
+library(data.table)
+
+# read table as per Q for mortals 
+atomicTypesTable <- fread('
+Type,Size,CharType,NumType,Notation,Null Value
+boolean,1,b,1,1b,0b                                                                                 
+byte,1,X,4,0x26,0x00                                                                                
+short,2,H,5,42h,0Nh                                                                                 
+int,4,I,6,42,0N                                                                                     
+long,8,J,7,42j,0Nj                                                                                  
+real,4,E,8,4.2e,0Ne                                                                                 
+float,8,F,9,4.2,0n                                                                                  
+char,1,C,10,"z"," "                                                                                 
+symbol,*,S,11,`zaphod,`                                                                             
+timestamp,8,P,12,2015.01.01T00:00:00.000000000,0Np                                                  
+month,4,M,13,2006.07m,0Nm                                                                           
+date,4,D,14,2006.07.21,0Nd                                                                          
+(datetime),4,Z,15,2006.07.21T09:13:39,0Nz                                                           
+timespan,8,N,16,12:00:00.000000000,0Nn                                                              
+minute,4,U,17,23:59,0Nu                                                                             
+second,4,V,18,23:59:59,0Nv                                                                          
+time,4,T,19,09:01:02:042,0Nt                                                                        
+enumeration,,,20+,`sym$`kx,                                                                         
+table,,,98,([] c1:`a`b`c; c2:10 20 30),                                                             
+dictionary,,,99,`a`b`v!10 20 30,                                                                    
+function,,,100,{x},                                                                                 
+nil item,,,101,::,', header=TRUE)
+```
+
+```{r atoms 1, echo=FALSE, warning=FALSE, message=FALSE}
+knitr::kable(atomicTypesTable)
+```
+
+The following function allows to check how kdb atomic objects are returned in R:
+
+```{r atoms 2, echo=TRUE}
+test <- function(h, type, x){
```

Comment:
instead of readme could you please put this tests into `tests/testthat/test-basic.R`, so I can integrate them into testing of the package?